File: problem/Smallest_Missing_Integer/findSmallestInteger.js
/**
 * @param {number[]} nums
 * @param {number} value
 * @return {number}
 */
var findSmallestInteger = function(nums, v) {
    let n = nums.length, res = 0;
    let rem = new Array(v).fill(0);
    for (let i = 0; i < n; i++) {
        let x = ((nums[i] % v) + v) % v;
        rem[x]++;
    }
    while (rem[res % v]--) res++;
    return res;
};

File: problem/Smallest_Missing_Integer/findSmallestInteger.c
int findSmallestInteger(int *nums, int numsSize, int v)
{
    int n = numsSize, res = 0;
    int rem[v];
    for (int i = 0; i < v; i++)
        rem[i] = 0;
    for (int i = 0; i < n; i++)
    {
        int x = ((nums[i] % v) + v) % v;
        rem[x]++;
    }
    while (rem[res % v]--)
        res++;
    return res;
}